mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -213,31 +197,20 @@ def _suggest(
         pass  # pylint: disable=unnecessary-pass # pragma: no cover
 
     @abstractmethod
-    def register_pending(
-        self,
-        *,
-        configs: pd.DataFrame,
-        context: Optional[pd.DataFrame] = None,
-        metadata: Optional[pd.DataFrame] = None,
-    ) -> None:
+    def register_pending(self, *, pending: Suggestion) -> None:
         """
-        Registers the given configs as "pending". That is it say, it has been suggested
-        by the optimizer, and an experiment trial has been started. This can be useful
-        for executing multiple trials in parallel, retry logic, etc.
+        Registers the given suggestion as "pending". That is it say, it has been
+        suggested by the optimizer, and an experiment trial has been started. This can
+        be useful for executing multiple trials in parallel, retry logic, etc.
 
         Parameters
         ----------
-        configs : pd.DataFrame
-            Dataframe of configs / parameters. The columns are parameter names and
-            the rows are the configs.
-        context : pd.DataFrame
-            Not Yet Implemented.
-        metadata : Optional[pd.DataFrame]
-            Not Yet Implemented.
+        pending: Suggestion
+            The pending suggestion to register.
         """
         pass  # pylint: disable=unnecessary-pass # pragma: no cover
 
-    def get_observations(self) -> Tuple[pd.DataFrame, pd.DataFrame, Optional[pd.DataFrame]]:
+    def get_observations(self) -> Observation:
```

Comment:
```suggestion
    def get_observations(self) -> Observations:
```

docstring needs updating too